cabal-install/src/Distribution/Client/Init/Utils.hs

```diff
@@ -277,11 +276,28 @@ chooseDep v flags (importer, m, mipi) = case mipi of
       Flag x -> x                   < CabalSpecV2_0
       NoFlag -> defaultCabalVersion < CabalSpecV2_0
 
-filePathToPkgName :: FilePath -> P.PackageName
-filePathToPkgName = PN.mkPackageName . Prelude.last . splitDirectories
+filePathToPkgName :: Interactive m => FilePath -> m P.PackageName
+filePathToPkgName = fmap (mkPackageName . repair . fromMaybe "" . safeLast . splitDirectories)
```

Comment:
I understand this is not your code, just shuffled around, but does the user get a file/line/col indication if `fromMaybe ""` explodes?

Reply:
fromMaybe won't explode: it's not fromJust. Instead, it will return the default value "", which will be dealt with inside repair, I think (I haven't checked, but I think it will).

Reply:
precisely @ulysses4ever - this will be safe